rules/CodeQuality/Rector/Class_/NarrowUnusedSetUpDefinedPropertyRector.php

```diff
@@ -0,0 +1,183 @@
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\NodeFinder;
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\MethodName;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\CodeQuality\Rector\Class_\NarrowUnusedSetUpDefinedPropertyRector\NarrowUnusedSetUpDefinedPropertyRectorTest
+ */
+final class NarrowUnusedSetUpDefinedPropertyRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const MOCK_OBJECT_CLASS = 'PHPUnit\Framework\MockObject\MockObject';
+
+    private readonly NodeFinder $nodeFinder;
+
+    public function __construct(
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer,
+    ) {
+        $this->nodeFinder = new NodeFinder();
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Turn property used only in setUp() to variable', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use PHPUnit\Framework\TestCase;
+
+class SomeServiceTest extends TestCase
+{
+    private $someServiceMock;
+
+    public function setUp(): void
+    {
+        $this->someServiceMock = $this->createMock(SomeService::class);
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use PHPUnit\Framework\TestCase;
+
+class SomeServiceTest extends TestCase
+{
+    public function setUp(): void
+    {
+        $someServiceMock = $this->createMock(SomeService::class);
+    }
+}
+CODE_SAMPLE
+                ,
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->testsNodeAnalyzer->isInTestClass($node)) {
+            return null;
+        }
+
+        $setUpClassMethod = $node->getMethod(MethodName::SET_UP);
+        if (! $setUpClassMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        $hasChanged = false;
+
+        foreach ($node->stmts as $key => $classStmt) {
+            if (! $classStmt instanceof Property) {
+                continue;
+            }
+
+            $property = $classStmt;
+            if (! $property->type instanceof Name) {
+                continue;
+            }
+
+            if (! $this->isName($property->type, self::MOCK_OBJECT_CLASS)) {
+                continue;
+            }
+
+            if (! $this->isPropertyUsedOutsideSetUpClassMethod($node, $setUpClassMethod, $property)) {
+                continue;
+            }
```

Comment:
the check is flipped, continue should check when property is used

Reply:
ok, it seems fixed at https://github.com/rectorphp/rector-phpunit/commit/f24bef525133df61ef556e34965f45625c8215c6